src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.Accessor.cs

```diff
@@ -135,7 +135,9 @@ private bool WriteStream(TKey key, Stream stream, Checksum? checksum, Cancellati
                         // Write the information into the in-memory write-cache.  Later on a background task
                         // will move it from the in-memory cache to the on-disk db in a bulk transaction.
                         InsertOrReplaceBlobIntoWriteCache(
-                            connection, dataId, checksumBytes, checksumLength, dataBytes, dataLength);
+                            connection, dataId,
+                            new ReadOnlySpan<byte>(checksumBytes, 0, checksumLength),
+                            new ReadOnlySpan<byte>(dataBytes, 0, dataLength));
```

Comment:
❔ Can we not obtain this directly from the `checksum` instance? If it's not there, we could still use `stackalloc` to produce a `Span<byte>` instead of operating against pools.